test/fuzz/simplify.cpp

```diff
@@ -134,60 +132,60 @@ Expr random_expr(Type T, int depth, bool overflow_undef) {
     }
 
     if (depth-- <= 0) {
-        return random_leaf(T, overflow_undef);
+        return random_leaf(fdp, T, overflow_undef);
     }
 
     const int bin_op_count = sizeof(make_bin_op) / sizeof(make_bin_op[0]);
     const int bool_bin_op_count = sizeof(make_bool_bin_op) / sizeof(make_bool_bin_op[0]);
     const int op_count = bin_op_count + bool_bin_op_count + 5;
 
-    int op = rng() % op_count;
+    int op = fdp.ConsumeIntegralInRange<int>(0, op_count);
```

Comment:
Should be `fdp.ConsumeIntegralInRange<int>(0, op_count-1)`

Reply:
(I realize the fragility was pre-existing before your changes, but improving the resilience here seems easy and worthwhile while we are here)

Reply:
Yeah, looking at this again a second time I'm struggling to understand what is going on here. Where did [this `5`](https://github.com/halide/Halide/pull/7512/files#diff-0bb69bc3a0270944620451c6da8561235b4d1062aa5acea5357e3aa29f99e451R140) come from.